content/influxdb/v2.0/monitor-alert/custom-checks.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Create custom checks
+seotitle: Custom checks
+description: >
+  Create custom checks with a Flux task.
+menu:
+  influxdb_2_0:
+    parent: Monitor & alert
+weight: 201
+v2.0/tags: [alerts, checks, tasks, Flux]
+---
+
+In the UI, you can create two kinds of [checks](/influxdb/v2.0/reference/glossary/#check):
+[`Threshold check`](/influxdb/v2.0/monitor-alert/checks/create/#threshold-check) and
+[`Deadman check`](/influxdb/v2.0/monitor-alert/checks/create/#deadman-check).
+
+Using a Flux task, you can create a custom check that provides a couple advantages:
+
+- Set up custom criteria for your alert (other than `threshold` and `deadman`).
+- Share custom checks across distributed instances.
+
+## Create a task
+
+1. In the InfluxDB UI, select **Tasks** in the navigation menu on the left.
+
+    {{< nav-icon "tasks" >}}
+
+2. Click **{{< icon "plus" >}} Create Task**, and then select **New Task**.
+3. In the **Name** field, enter a descriptive name,
+   and then enter how often to run the task in the **Every** field (for example, `10m`).
+   For more detail, such as using cron syntax or including an offset, see [Task configuration options](/influxdb/v2.0/process-data/task-options/).
+4. Enter the Flux script for your custom check, including the [`monitor.check`](/influxdb/v2.0/reference/flux/stdlib/monitor/check/) function.
+
+### Example: Monitor failed tasks
+
+Receive an alert when a task fails.
+
+The script below is fairly complex, and can be used an a framework for similar tasks.
+It does the following:
+
+- Import the necessary `influxdata/influxdb/monitor` package, and other packages for data processing.
+- Query the `_tasks` bucket to retrieve all statuses generated by your check.
+- Set the `_level` to alert on, for example, `crit`, `warn`, `info`, or `ok`.
+- Create a `check` object that specifies an ID, name, and type for the check.
+- Define the `ok` and `crit` statuses.
+- Execute the `monitor` function on the `check` using the `task_data`.
+
+#### Example alert task script
+
+```js
+import "strings"
+import "regexp"
+import "influxdata/influxdb/monitor"
+import "influxdata/influxdb/v1"
+
+option task = {name: "Failed Tasks Check", every: 1h, offset: 4m}
+
+task_data = from(bucket: "_tasks")
+	|> range(start: -1h, stop: now())
+	|> filter(fn: (r) =>
+		(r["_measurement"] == "runs"))
+	|> filter(fn: (r) =>
+		(r["_field"] == "logs"))
+	|> map(fn: (r) => ({ r with name: strings.split(v: regexp.findString(r: /option task = \{([^\}]+)/, v: r._value), t: "\\\\\\\"")[1] }))
+	|> drop(columns: ["_value", "_start", "_stop"])
+	|> group(columns: ["name", "taskID", "status", "_measurement"])
+	|> map(fn: (r) =>
+		({r with _value: if r.status == "failed" then 1 else 0}))
+	|> last()
+check = {
+	_check_id: "0000000000000001",
+	_check_name: "Failed Tasks Check",
+	_type: "custom",
+	tags: {},
```

Comment:
Perhaps we could add comments (or clarify on line 44):
- check id: "16 alphanumeric characters (letters must be lowercase)"
- check name: "string"
- type: point out here you could also use deadman or threshold